src/main/sensors/compass.c

```diff
@@ -349,6 +355,16 @@ bool compassInit(void)
 
     buildRotationMatrixFromAlignment(&compassConfig()->mag_customAlignment, &magDev.rotationMatrix);
 
+    if (magDev.magOdrHz) {
+        // For Mags that send data at a fixed ODR, we wait some quiet period after a read before checking for new data
```

Comment:
See above. in single measurement mode, no check for data is done, but readout delay musty be long enough

Reply:
Thanks, I will need to handle the very different requirements of single measurement modes.